packages/embed/src/factories/create-popover/create-popover.ts

```diff
@@ -97,6 +97,48 @@ const buildTooltip = (message: string, hide: () => void) => {
   return container
 }
 
+const LOCAL_STORAGE_NOTIFICATION_DATA = 'tfNotificationData'
+
+type LocalStorageCUIData = {
+  formId: string
+  expireDate: number
+}
+const today = new Date().getTime()
+
+const cuiLocaslStorageDate = typeof window !== 'undefined' && localStorage.getItem(LOCAL_STORAGE_NOTIFICATION_DATA)
+
+const getLocalStorageData = () => cuiLocaslStorageDate && JSON.parse(cuiLocaslStorageDate)
+
+function saveToLocalstorage(data: LocalStorageCUIData) {
+  if (!data) return null
+  typeof window != 'undefined' && localStorage.setItem(LOCAL_STORAGE_NOTIFICATION_DATA, JSON.stringify(data))
+}
+
+function canBuildCUIDot(formId: string, days: number) {
```

Comment:
Its just a dot, can work for both CUI or form. It is not coupled to rendered interface:
```suggestion
function canBuildNotificationDot(formId: string, days: number) {
```